api/api.go

```diff
@@ -63,11 +64,21 @@ type CommitResponse struct {
 	Warnings []string    `zed:"warnings"`
 }
 
-type IndexPostRequest struct {
-	Keys     []string `json:"keys"`
-	Name     string   `json:"name"`
-	Patterns []string `json:"patterns"`
-	Zed      string   `json:"zed,omitempty"`
+type IndexRulesDeleteRequest struct {
+	RuleIDs []string `zed:"rule_ids"`
+}
+
+type IndexRulesDeleteResponse struct {
+	Rules []index.Rule `zed:"rules"`
+}
+
+type ApplyIndexRequest struct {
+	RuleName string        `zed:"rule_name"`
+	Tags     []ksuid.KSUID `zed:"tags"`
+}
+
+type UpdateIndexRequest struct {
```

Comment:
Maybe call these IndexApplyRequest and IndexUpdateRequest so all the index-related types have a common prefix?